### ISSUE TITLE
Composed functions to the left don't match hints

### ISSUE THREAD

**Issue by ndmitchell** [ndmitchell/hlint#890]
Given the `isLeft (fmap f x) ==> isLeft x` hint, it should trigger in all these 4 cases, but it only triggers in the first three.

```
foo f x = isLeft (fmap f x)
bar f = isLeft . fmap f
baz f x = isLeft $ fmap f x
qux magic f x = magic . isLeft $ fmap f x
```

From #882
